app/controllers/finance/api/payment_callbacks/stripe_callbacks_controller.rb

```diff
@@ -21,7 +21,14 @@ class MissingStripeEndpointSecret < StripeCallbackError; end
 
   # Undocumented endpoint used for update callbacks of async-authorized payment transactions (mostly due to SCA regulations)
   def create
-    service = Finance::StripePaymentIntentUpdateService.new(current_account, stripe_event)
+    begin
+      service = Finance::StripePaymentIntentUpdateService.new(current_account, stripe_event)
+    rescue ActiveRecord::RecordNotFound
+      # Returning 204 to acknowledge reception even for payments we don't care about
+      # https://issues.redhat.com/browse/THREESCALE-6851
+      Rails.logger.info("Acknowledging reception: '#{stripe_event.id}'")
```

Comment:
As we are not caring about these event, I am not sure we should even print this message in the log.
Or, if we do, I would provide more information, because it would be way too cryptic, I think (reception of what? there would be just some random ID).
I'd probably opt for not printing anything.

In fact, I am not sure we need to wait the result from `service.call` to return `204`...
That's what Stripe says about it: https://stripe.com/docs/webhooks#acknowledge-events-immediately

> Your endpoint must quickly return a successful status code (`2xx`) prior to any complex logic that could cause a timeout. For example, you must return a `200` response before updating a customer’s invoice as paid in your accounting system.

I'd say as soon as we populate `stripe_event` successfully, we can return `204`. What do you think @jlledom ?

Reply:
Then we should call `service.call` in a a background job or another thread? and who is receiving the exception in the case it fails?

Reply:
@jlledom I think for now we can leave it as it is. This controller is not known for causing troubles (except for the issue that is fixed in this PR), so probably we don't need to complicate the things, and we can leave it as it is.